### PR TITLE
Add test_parallel_core_flushing

### PR DESCRIPTION
**Pull request by anikielx**
Signed-off-by: Adriana Nikelska <adrianax.nikielska@intel.com>

### CODE REVIEW

**Comment by mmichal10** [comment]
ok to test

**Comment by bkarpins** [comment]
retest this please

**Comment by bkarpins** [comment]
retest this please

**Comment by bkarpins** [comment]
retest this please

File: test/functional/tests/lazy_writes/test_parallel_core_flushing.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    with TestRun.step("Check whether all CAS devices are in 'Flushing' state and wait for finish."):
+        for core in cores:
+            wait_for_flushing_start(cache, core)
+
+        percentages = [0] * len(cores)
+        try:
+            while not all(p == 100 for p in percentages):
+                for i, core in enumerate(cores):
+                    if percentages[i] != 100:
+                        percentage = get_flushing_progress(cache.cache_id, core.core_id)
```
Comment (mmichal10) [inline test/functional/tests/lazy_writes/test_parallel_core_flushing.py:80]: As soon as one of cores is flushed, `get_flushing_progress()` will throw an exception. The loop will be interrupted and the flushing progress bar for other cores won't be tested. I suggest putting only `get_flushing_progress()` into `try` block.

Comment (mmichal10) [reply test/functional/tests/lazy_writes/test_parallel_core_flushing.py:80]: You can even go one step further and implement function that would return a list with flushing progress for each core. 

```
def get_progress(cache, cores):
    progress = [0] * len(cores)
    for i, core in enumerate(cores)::
        try:
            progress[i] = get_flushing_progress(cache.cache_id, core.core_id)
        except CmdException:
             progress[i] = 100
    return progress
```

usage:

```
percentages = [0] * len(cores)
while TestRun.executor.check_if_process_exists(pid):
    current_values = get_progress(cache, cores)

    for old, new, core zip(percentages, current_values, cores)
        if old > new:
            TestRun.fail(f"Core {core.id}: progress decreased from {old} to {new}")

    percentage = current_values
```

**Review by mmichal10 — APPROVED** [review]

**Comment by mmichal10** [comment]
@anikielx please update license headers

**Review by mmichal10 — APPROVED** [review]

**Review by katlapinka — APPROVED** [review]

**Pull request merged**